zookeeper-docs/src/main/resources/markdown/zookeeperUseCases.md

```diff
@@ -0,0 +1,377 @@
+    - Ensuring the generation of unique values in various sequence ID generators
+    - Cassandra backups
+    - Implementing the TrackID service
+    - Performing leader election with ZooKeeper for various distributed tasks
+    - Implementing a distributed semaphore for concurrent jobs
+    - Distributed caching
+
+### [Nutanix](http://www.nutanix.com/)
+  - Nutanix develops a hyper-converged storage and compute solution, which leverages local components of a host machine, such as storage capacity (disks) and compute (CPUs),
+    to create a distributed platform for virtualization. This solution is known as the Nutanix Virtual Computing Platform. It supports industry-standard hypervisor ESXi, KVM, and Hyper-V.
+    The platform is bundled in the appliance form factor with two nodes or four nodes. A VM in the platform known as the Nutanix Controller VM works as the decision subsystem, which manages the platform.
+  - The Nutanix platform uses Apache ZooKeeper as a Cluster configuration manager. The configuration data that pertains to the platform, such as hostnames, IP addresses,
+    and the cluster state, is stored in a ZooKeeper ensemble. ZooKeeper is also used to query the state of the various services that run in the platform[21].
+
+### [Pinterest](https://www.pinterest.com/)
+ - Pinterest uses the ZooKeeper for Service discovery and dynamic configuration.Like many large scale web sites, Pinterest’s infrastructure consists of servers that communicate with
+   backend services composed of a number of individual servers for managing load and fault tolerance. Ideally, we’d like the configuration to reflect only the active hosts,
+   so clients don’t need to deal with bad hosts as often. ZooKeeper provides a well known pattern to solve this problem[19].
+
+### [Rackspace](http://www.rackspace.com/email_hosting)
+ - The Email & Apps team uses ZooKeeper to coordinate sharding and responsibility changes in a distributed e-mail client
+   that pulls and indexes data for search. ZooKeeper also provides distributed locking for connections to prevent a cluster from overwhelming servers.
+
+### [Sematext](http://sematext.com/)
+ - Uses ZooKeeper in SPM (which includes ZooKeeper monitoring component, too!), Search Analytics, and Logsene.
+
+### [Tubemogul](http://tubemogul.com/)
+ - Uses ZooKeeper for leader election, configuration management, locking, group membership
+
+### [Twitter](https://twitter.com/)
+ - ZooKeeper is used at Twitter as the source of truth for storing critical metadata. It serves as a coordination kernel to
+   provide distributed coordination services, such as leader election and distributed locking.
+   Some concrete examples of ZooKeeper in action include[15,16]:
+   - ZooKeeper is used to store service registry, which is used by Twitter’s naming service for service discovery.
+   - Manhattan, Twitter’s in-house key-value database, stores its cluster topology information in ZooKeeper.
```

Comment:
I'd like to add a little bit more content here:
`Manhattan (Twitter’s in-house key-value database), Nighthawk (sharded Redis), and Blobstore (in-house photo and video storage),  store their cluster topology information in ZooKeeper.`